ted/src/main/java/frc/robot/subsystems/LEDSubsystem.java

```diff
@@ -0,0 +1,166 @@
+              targetLedState = LEDState.Yellow;
+        }
+        else if(currentActions.containsKey(LEDState.OrangeSolid) && currentActions.get(LEDState.OrangeSolid).booleanValue()) {
+              targetLedState = LEDState.OrangeSolid;
+        }
+        else if(currentActions.containsKey(LEDState.OrangeBlink) && currentActions.get(LEDState.OrangeBlink).booleanValue()) {
+              targetLedState = LEDState.OrangeBlink;
+        }
+        // update the LED state when the target state has changed
+        if(this.currentLEDState != targetLedState) {
+              this.currentLEDState = targetLedState;
+              if(this.currentLEDState == LEDState.Green) {
+                    this.greenSolid();
+              }
+              else if(this.currentLEDState == LEDState.Yellow) {
+                    this.yellowSolid();
+              }
+              else if(this.currentLEDState == LEDState.OrangeSolid) {
+                    this.orangeSolid();
+              }
+              else if(this.currentLEDState == LEDState.OrangeBlink) {
+                    this.orangeBlink();
+              }
+              else if(this.currentLEDState == LEDState.Off) {
+                    this.offState();
+              }
+              System.out.println("**** UPDATING LED STATE TO " + this.currentLEDState.toString());
+        }
+        else if(this.lastBlinkState != this.currentBlinkState && this.currentLEDState == LEDState.OrangeBlink) {
+              System.out.println("**** BLINKING LED STATE TO " + this.currentLEDState.toString());
+              this.orangeBlink();
+        }
+  }
+
+      //Sets leds to orange blink
+      private void orangeBlink() {
+            if(this.currentBlinkState) {
+                  this.setLedStringColor(255,165,0);
+            }
+            else {
+                  this.setLedStringColor(0, 0, 0);
+            }
+      }
+
+      //Sets leds to orange solid
+      private void orangeSolid() {
+            this.setLedStringColor(255,165,0);
+      }
+
+      //Sets leds to yellow solid
+      private void yellowSolid() {
+            this.setLedStringColor(150,150,0);
+      }
+
+      //Sets leds to green solid
+      private void greenSolid() {
+            this.setLedStringColor(0,200,0);
+      }
+
+      //Turns off leds
+      private void offState() {
+            this.setLedStringColor(0,0,0);
+            this.leds.modulateVBatOutput(0);
+      }
+
+      //Sets leds color
+      private void setLedStringColor(int red, int green, int blue) {
+                  this.leds.setLEDs(red, green, blue, 0, startIdx, BUFFER_LENGTH);
+                  this.leds.modulateVBatOutput(0.9);
+      }
+
+      //Updates the blink state of leds
+      private void updateBlinkCounterState() {
+            this.lastBlinkState = this.currentBlinkState;
+            this.blinkCounter++;
+            if(this.blinkCounter % this.blinkPeriodInHertz == 0){
```

Comment:
I don't think this math works out to the requested blink frequency. I believe periodic is called once per tick (20ms) or 50 times per second. So, create a class variable for ticksPerSecond = 50, and then this should be this.blinkCount % (ticksPerSecond / thisBlinkPeriodInHz)
e.g. 25Hz would blink every 2 ticks. 
You probably also want to update your blink frequency variable to be 2 rather than 25.